nifi-nar-bundles/nifi-oraclecdc-bundle/nifi-oraclecdc-processors/src/main/java/org/apache/nifi/processors/oraclecdc/controller/impl/StandardOracleCDCService.java

```diff
@@ -0,0 +1,269 @@
+        dataSource.setMaxIdle(maxTotal);
+        // dataSource.setMaxConnLifetimeMillis(maxConnLifetimeMillis);
+        // dataSource.setTimeBetweenEvictionRunsMillis(timeBetweenEvictionRunsMillis);
+        // dataSource.setMinEvictableIdleTimeMillis(minEvictableIdleTimeMillis);
+        // dataSource.setSoftMinEvictableIdleTimeMillis(softMinEvictableIdleTimeMillis);
+
+        dataSource.setUrl(dburl);
+        dataSource.setUsername(user);
+        dataSource.setPassword(passw);
+
+        context.getProperties().keySet().stream().filter(PropertyDescriptor::isDynamic)
+                .forEach((dynamicPropDescriptor) -> dataSource.addConnectionProperty(dynamicPropDescriptor.getName(),
+                        context.getProperty(dynamicPropDescriptor).evaluateAttributeExpressions().getValue()));
+
+    }
+
+    @Override
+    public void receiveEvents(Object xsOut, OracleCDCEventHandler handler) {
+        if (null == xsOut) {
+            getLogger().info("xstreamOut is null");
+            return;
+        }
+
+        try {
+            LCRCallBackHandler hdlr = new LCRCallBackHandler(this.driverClassLoader, handler);
+            Object proxy = Proxy.newProxyInstance(this.driverClassLoader,
+                    new Class[] { loadClass("oracle.streams.XStreamLCRCallbackHandler") }, hdlr);
+            Class<?> xstreamOut = loadClass("oracle.streams.XStreamOut");
+            Method method = xstreamOut.getMethod("receiveLCRCallback",
+                    loadClass("oracle.streams.XStreamLCRCallbackHandler"), int.class);
+            method.invoke(xsOut, proxy,
+                    loadClass("oracle.streams.XStreamOut").getDeclaredField("DEFAULT_MODE").getInt(null));
+            System.out.print("done");
+        } catch (Exception e) {
+            e.printStackTrace();
+            getLogger().warn("exception when processing LCRs");
+            getLogger().warn(e.getMessage());
+            throw new ProcessException("exception when processing LCRs " + e.getMessage());
+        }
+    }
+
+    @Override
+    public void setProcessedLowWaterMark(Object xsOutServer, byte[] position) {
+        try {
+            System.out.println(new String(new Base32(true).encode(position)));
+            Class<?> xstreamOut = loadClass("oracle.streams.XStreamOut");
+            Method method = xstreamOut.getMethod("setProcessedLowWatermark", byte[].class, int.class);
+            method.invoke(xsOutServer, position,
+                    loadClass("oracle.streams.XStreamOut").getDeclaredField("DEFAULT_MODE").getInt(null));
+        } catch (Exception e) {
+            e.printStackTrace();
+            getLogger().warn("exception when set low watermark");
+            getLogger().warn(e.getMessage());
+            throw new ProcessException("exception when setting low water mark " + e.getMessage());
+        }
+    }
+
+    @Override
+    public Object attach(String xsOutName, byte[] lastPosition) {
+        getLogger().info("in attach");
```

Comment:
This should probably be debug or trace level log (it it's still needed.)

Reply:
will do the change